File: TopCoder04/PHP/01/Main.php
<?php
    class InterestingDigits {
    	public function digits($base)
		{
			$v = array();
			
			for($n = 2; $n < $base ; $n++){
				$ok = true;
				for($f1 = 0; $f1 < $base ; $f1++){
					for($f2 = 0; $f2< $base; $f2++){
						for($f3 = 0; $f3 <$base; $f3++){
							if(($f1 + $f2*$base + $f3*$base*$base) % $n == 0 && ($f1+$f2+$f3) % $n != 0){
								$ok = false;
								break 3;
							}
						}
					}
				}
				if($ok){
					$v[] = $n;	
				}

			}
			return $v;
		}
    }
	
	$base = 10;
	$id = new InterestingDigits;
	$result = $id->digits($base);
	
	foreach ($result as $value) {
		echo $value ," ";
	}
	
?>

File: TopCoder04/Ruby/01/Main.rb
class InterestingDigits
  def dogots(base)
    ans = []
    for n in 2...base
      ok = true
      for k1 in 0...base
        for k2 in 0...base
          for k3 in 0...base
            if (k1+k2*base+k3*base*base) % n == 0 and (k1+k2+k3) % n != 0
              ok = false
              break
            end
          end
          if !ok
            break
          end
        end
        if !ok
          break
        end
      end
      if ok
        ans << n
      end
    end
    return ans 
  end 
end

puts InterestingDigits.new.dogots(10)
